### DISCUSSION TITLE
What's the difference between this repo and creating a new HardHat typescript project?

### DISCUSSION THREAD

**Discussion by martin-opensky** [PaulRBerg/hardhat-template#108] [General]
Hello everyone,

I've been following this project for the past 2 months as well as working locally with Hardhat.  I noticed when I created a new project with the Hardhat CLI, and select TypeScript, that the packages included were practically identical to the ones in this project.  
This makes me wonder: Is there anything that this solidity template offers that the current create new Hardhat typescript project doesn't?

Thank you

**Comment by PaulRBerg** [comment]
Hi @martin-opensky! 

> when I created a new project with the Hardhat CLI, and select TypeScript, that the packages included were practically identical to the ones in this project

Well that certainly wasn't the case when I first created this template.

> Is there anything that this solidity template offers that the current create new Hardhat typescript project doesn't?

There are many different things:

- This template uses yarn v3 (a.k.a. berry), while Hardhat CLI does not.
- This template provides an (opinionated) list of Prettier rules.
- This template provides more (opinionated) Solhint rules.
- This template provides a [task](https://github.com/paulrberg/solidity-template/blob/b173741333327fc60c48317c0582e127f7765e97/tasks/deploy/greeter.ts) for deploying the `Greeter` contract.
- This template provides network configurations for Arbitrum, Avalanche, BSC, Ethereum, Optimism, Polygon Mainnet, Polygon Mumbai, and Rinkeby. The Hardhat CLI provides only one network (Ropsten).
- This template has different ignore files (`.gitignore`, `.prettierignore`, and `.solhintignore`).

And I could go and on. You're just scratching the surface by looking at the dependencies listed in `package.json`.

**Comment by martin-opensky** [comment]
Hey @paulrberg,

Thanks for your quick and thorough response.  I thought I was missing something, so appreciate your answer :-)

**Reply by PaulRBerg** [reply]
No prob.
